leaders/src/main/java/edu/cornell/partitioner/methods/NumSplitPartitionMethod.java

```diff
@@ -0,0 +1,48 @@
+package edu.cornell.partitioner.methods;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Provides method that divides classes equally among a set number of partitions
+ */
+public class NumSplitPartitionMethod extends PartitionMethod {
+
+    /**
+     * Split the list of classes into equal number of input partitions. <br></br> If lists cannot be divided evenly among
+     * partitions, include one extra test in each test set up until the modulus of className size and partitions.
+     * @param classNames - List of the classes by package+name
+     * @param partitions - Number of partitions to create
+     * @return A list of approximately evenly split test class name partitions stored as sets
+     */
+    @Override
+    public List<Set<String>> partition(List<String> classNames, int partitions) {
+        List<Set<String>> classes = new ArrayList<>();
+
+        int listSize = classNames.size();
+
+        int startIndex = 0;
+        int overflow = listSize % partitions;
+        int endIndex;
+
+        for (int i = 0; i < partitions; i++) {
```

Comment:
You could probably use `i` to keep track of the start and end index with some calculations instead of having separate variables and logic.

Reply:
@owenxr For you

Reply:
What makes it difficult is the overflow. I will investigate.

Reply:
I suggest we leave this in for now but we can refactor later (in the interest of time)

Reply:
I agree. Not sure if there is much benefit as there may be some duplicate calculations and the complexity of determining the previous iterations overflow with edge cases. At least, I haven't been able to find a way to do this concisely.